src/node/nodestate.h

```diff
@@ -205,6 +205,72 @@ namespace ccf
       sm.advance(State::initialized);
     }
 
+    std::vector<uint8_t> serialize_genesis(
+      const CreateNew::In& args, std::vector<uint8_t>& quote)
+    {
+      jsonrpc::ProcedureCall<CreateNetworkNodeToNode::In> create_rpc;
+
+      create_rpc.id = 0;
+      create_rpc.method = ccf::MemberProcs::CREATE;
+
+      for (auto& cert : args.config.genesis.member_certs)
+      {
+        create_rpc.params.member_cert.push_back(cert);
+      }
+
+      create_rpc.params.gov_script = args.config.genesis.gov_script;
+      create_rpc.params.node_cert = node_cert;
+      create_rpc.params.network_cert = network.secrets->get_current().cert;
+      create_rpc.params.quote = quote;
+      create_rpc.params.code_digest =
+        std::vector<uint8_t>(std::begin(node_code_id), std::end(node_code_id));
+      create_rpc.params.node_info_network = args.config.node_info_network;
+
+      nlohmann::json j = create_rpc;
+      auto contents = nlohmann::json::to_msgpack(j);
+
+      auto sig_contents = node_kp->sign(contents);
+
+      nlohmann::json sj;
+      sj["req"] = j;
+      sj["sig"] = sig_contents;
+
+      return jsonrpc::pack(sj, jsonrpc::Pack::Text);
+    }
+
+    void send_request(std::vector<uint8_t>& packed)
```

Comment:
This is a nitpick, but `send_create_request`? if we are only every going to use this to send the create request and no other